plugins/MumblePluginProcess.hpp

```diff
@@ -0,0 +1,313 @@
+// Copyright 2020 The Mumble Developers. All rights reserved.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file at the root of the
+// Mumble source tree or at <https://www.mumble.info/LICENSE>.
+
+#ifndef MUMBLEPLUGINPROCESS_HPP_
+#define MUMBLEPLUGINPROCESS_HPP_
+
+#include "mumble_plugin_utils.h"
+
+#include <chrono>
+#include <cstdint>
+#include <cstring>
+#include <map>
+#include <vector>
+
+#ifdef OS_LINUX
+#	include <sstream>
+
+#	include <sys/uio.h>
+#endif
+
+typedef uint32_t procid_t;
+typedef uint64_t procptr_t;
+
+/// Abstract class.
+/// Only defines stuff that can be used with both Linux and Windows processes.
+class MumblePluginProcess {
```

Comment:
I don't like using the term "plugin" when what it really is related to is "positional audio". Especially since there will be actual plugins coming to Mumble in the foreseeable future.

Reply:
I agree. Does `ProcessUtils` sound good?

Reply:
Yeah that would be okay. Depending on what exactly the class is meant to be responsible for, `Process` might be an appropriate name in and of itself :thinking:

Reply:
Anything process-related. Mostly memory reading.

Reply:
In that case I think I would even prefer `Process`, but `ProcessUtils` is fine as well. Your call :)